tests/http/test_http_copy.py

```diff
@@ -0,0 +1,104 @@
+import os
+import pytest
+import hashlib
+import time
+
+pytestmark = [
+    pytest.mark.disable_loganalyzer,
+    pytest.mark.topology("any"),
+    pytest.mark.device_type("vs"),
+]
+
+SONIC_SSH_PORT = 22
+SONIC_SSH_REGEX = "OpenSSH_[\\w\\.]+ Debian"
+HTTP_PORT = "8080"
+
+def test_http_copy(duthosts, rand_one_dut_hostname, ptfhost):
+    """Test that HTTP (copy) can be used to download objects to the DUT"""
+
+    duthost = duthosts[rand_one_dut_hostname]
+    ptf_ip = ptfhost.mgmt_ip
+
+    test_file_name = "test_file.bin"
+
+    # Copies http server file to ptf module_async=True
+    ptfhost.copy(src="http/start_http_server.py", dest="/tmp/start_http_server.py")
+    
+    # Starts the http server on the ptf
+    ptfhost.command("python /tmp/start_http_server.py", module_async=True)
+
+    # Validate HTTP Server has started
+    started = False
+    tries = 0
+    while not started and tries < 10:
+        if os.system("curl {}:8080".format(ptf_ip)) == 0:
+            started = True
+        tries += 1
+        time.sleep(1)
+
+    if not started:
+        pytest.fail("HTTP Server could not be started")
```

Comment:
We prefer use the pytest_assert to make the code compact. Again, you can reference tacacs test for example:

https://github.com/Azure/sonic-mgmt/blob/master/tests/tacacs/test_ro_user.py#L3

https://github.com/Azure/sonic-mgmt/blob/master/tests/tacacs/test_ro_user.py#L127

Sorry should have raised this with the other comments. Please address other places in this file below as well.

Reply:
Great, thanks again for the suggestion @yxieca, I hadn't seen that in the tests I was using for reference. I've just pushed the change.